### PR TITLE
Fix onRequest hook documentation

### PR DESCRIPTION
**Pull request by mattiapv**
Added missing `async` and `await` keywords, `try...catch` block. This prevents server crashes caused by missing `await` on `jwtVerify()` function.

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by mcollina — CHANGES_REQUESTED** [review]
None of this are needed, why did you add them?

**Comment by mattiapv** [comment]
> None of this are needed, why did you add them?

Because without awaiting `jwtVerify()` the server crash.

You can try the [cookie](https://github.com/fastify/fastify-jwt#cookie) or the [onlyCookie](https://github.com/fastify/fastify-jwt#cookie) example.

If you make a call to `http://localhost:3000/verifyCookie` the server crash.

If I add `await request.jwtVerify()`, I get back an error and the server doesn't crash.

File: README.md [hunk 1]
```diff
@@ -423,7 +439,13 @@ fastify.register(require('@fastify/jwt'), {
   secret: 'supersecret'
 });
 
-fastify.addHook('onRequest', (request, reply) =>  request.jwtVerify());
+fastify.addHook('onRequest', async (request) => {
```
Comment (Eomm) [inline README.md:442]: as before

File: README.md [hunk 2]
```diff
@@ -359,10 +365,14 @@ fastify.get('/cookies', async (request, reply) => {
...
 })
 
-fastify.addHook('onRequest', (request) => {
-    request.jwtVerify()
-    request.jwtVerify({onlyCookie: true})
-})
+fastify.addHook('onRequest', async (request) => {
+  try {
+    await request.jwtVerify()
+    await request.jwtVerify({onlyCookie: true})
```
Comment (Eomm) [inline README.md:371]: this seems a doc error:

it should be
```
(request) => {
  return request.jwtVerify({ onlyCookie: true })
}
```

File: README.md [hunk 3]
```diff
@@ -304,7 +304,13 @@ fastify.get('/cookies', async (request, reply) => {
...
 })
 
-fastify.addHook('onRequest', (request) => request.jwtVerify())
+fastify.addHook('onRequest', async (request) => {
+  try {
+    await request.jwtVerify();
+  } catch (err) {
+    reply.send(err);
+  }
+});
```
[the remark this answers is not in the document]
Comment (mattiapv) [reply README.md:313]: I would keep arrow function, my mistake sorry.

File: README.md (hunk 2 above)
Comment (mattiapv) [reply README.md:371]: I would it as

```js
fastify.addHook('onRequest', (request) =>
  request.jwtVerify({ onlyCookie: true })
);
```

**Review by Eomm — APPROVED** [review]

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**